File: ios/native_feed_player/Sources/native_feed_player/PlatformViewRegistry.swift
final class PlatformViewRegistry<Key: Hashable, View: AnyObject> {
  private var views: [Key: View] = [:]

  subscript(key: Key) -> View? {
    views[key]
  }

  func register(_ view: View, for key: Key) {
    views[key] = view
  }

  func removeIfCurrent(_ view: View, for key: Key) -> Bool {
    guard views[key] === view else {
      return false
    }
    views.removeValue(forKey: key)
    return true
  }

  func removeAll() {
    views.removeAll()
  }
}

File: android/src/main/kotlin/io/github/aswinsubhash/native_feed_player/PlatformViewRegistry.kt
package io.github.aswinsubhash.native_feed_player

internal class PlatformViewRegistry<Key, View : Any> {
    private val views = mutableMapOf<Key, View>()

    operator fun get(key: Key): View? = views[key]

    fun register(key: Key, view: View) {
        views[key] = view
    }

    fun removeIfCurrent(key: Key, view: View): Boolean {
        if (views[key] !== view) {
            return false
        }
        views.remove(key)
        return true
    }

    fun clear() {
        views.clear()
    }
}
